src/main/java/eu/europeana/metis/sandbox/service/workflow/TransformationServiceImpl.java

```diff
@@ -1,13 +1,48 @@
 package eu.europeana.metis.sandbox.service.workflow;
 
+import static java.util.Objects.requireNonNull;
+
+import eu.europeana.metis.sandbox.common.exception.RecordProcessingException;
 import eu.europeana.metis.sandbox.domain.Record;
+import eu.europeana.metis.transformation.service.EuropeanaIdCreator;
+import eu.europeana.metis.transformation.service.EuropeanaIdException;
+import eu.europeana.metis.transformation.service.TransformationException;
+import eu.europeana.metis.transformation.service.XsltTransformer;
+import java.nio.charset.StandardCharsets;
+import org.springframework.beans.factory.ObjectProvider;
 import org.springframework.stereotype.Service;
 
 @Service
 class TransformationServiceImpl implements TransformationService {
 
+  private final ObjectProvider<XsltTransformer> xsltTransformer;
+
+  public TransformationServiceImpl(
+      ObjectProvider<XsltTransformer> xsltTransformer) {
+    this.xsltTransformer = xsltTransformer;
+  }
+
   @Override
   public Record transform(Record record) {
-    return record;
+    requireNonNull(record, "Record must not be null");
+
+    String recordTransformed;
+    try {
+      var europeanaGeneratedIdsMap = new EuropeanaIdCreator()
+          .constructEuropeanaId(record.getContent(), record.getDatasetId());
+      var xsltTransformer = getTransformer(record.getDatasetName(),
+          record.getCountry().xmlValue(), record.getLanguage().xmlValue());
```

Comment:
The variable you're declaring here already exists as a class field. It's not good practice.